contracts/token/ERC865/ERC865Token.sol

```diff
@@ -0,0 +1,366 @@
+        return true;
+    }
+
+    /**
+     * @notice Increase the amount of tokens that an owner allowed to a spender.
+     * @param _signature bytes The signature, issued by the owner.
+     * @param _spender address The address which will spend the funds.
+     * @param _addedValue uint256 The amount of tokens to increase the allowance by.
+     * @param _fee uint256 The amount of tokens paid to msg.sender, by the owner.
+     * @param _nonce uint256 Presigned transaction number.
+     */
+    function increaseApprovalPreSigned(
+        bytes _signature,
+        address _spender,
+        uint256 _addedValue,
+        uint256 _fee,
+        uint256 _nonce
+    )
+        public
+        returns (bool)
+    {
+        require(_spender != address(0));
+        require(signatures[_signature] == false);
+
+        bytes32 hashedTx = increaseApprovalPreSignedHashing(address(this), _spender, _addedValue, _fee, _nonce);
+        address from = recover(hashedTx, _signature);
+        require(from != address(0));
+
+        allowed[from][_spender] = allowed[from][_spender].add(_addedValue);
+        balances[from] = balances[from].sub(_fee);
+        balances[msg.sender] = balances[msg.sender].add(_fee);
+        signatures[_signature] = true;
+
+        Approval(from, _spender, allowed[from][_spender]);
+        Transfer(from, msg.sender, _fee);
+        ApprovalPreSigned(from, _spender, msg.sender, allowed[from][_spender], _fee);
+        return true;
+    }
+
+    /**
+     * @notice Decrease the amount of tokens that an owner allowed to a spender.
+     * @param _signature bytes The signature, issued by the owner
+     * @param _spender address The address which will spend the funds.
+     * @param _subtractedValue uint256 The amount of tokens to decrease the allowance by.
+     * @param _fee uint256 The amount of tokens paid to msg.sender, by the owner.
+     * @param _nonce uint256 Presigned transaction number.
+     */
+    function decreaseApprovalPreSigned(
+        bytes _signature,
+        address _spender,
+        uint256 _subtractedValue,
+        uint256 _fee,
+        uint256 _nonce
+    )
+        public
+        returns (bool)
+    {
+        require(_spender != address(0));
+        require(signatures[_signature] == false);
+
+        bytes32 hashedTx = decreaseApprovalPreSignedHashing(address(this), _spender, _subtractedValue, _fee, _nonce);
+        address from = recover(hashedTx, _signature);
+        require(from != address(0));
+
+        uint oldValue = allowed[from][_spender];
+        if (_subtractedValue > oldValue) {
+            allowed[from][_spender] = 0;
+        } else {
+            allowed[from][_spender] = oldValue.sub(_subtractedValue);
+        }
+        balances[from] = balances[from].sub(_fee);
+        balances[msg.sender] = balances[msg.sender].add(_fee);
+        signatures[_signature] = true;
+
+        Approval(from, _spender, _subtractedValue);
```

Comment:
I guess `Approval(from, _spender, allowed[from][_spender]);` is correct (not `_subtractedValue`).